src/script/cpp_api/s_security.cpp

```diff
@@ -503,6 +504,133 @@ bool ScriptApiSecurity::safeLoadFile(lua_State *L, const char *path, const char
 	return result;
 }
 
+std::string inline _get_mod_name_from_debug(lua_Debug *info)
+{
+	std::string mod_name;
+	std::string src = info->source;
+	std::string BIN_DIR = std::string(DIR_DELIM) + "bin" + DIR_DELIM;
+	std::string MOD_DIR = BIN_DIR + ".." + DIR_DELIM + "mods" + DIR_DELIM;
+
+	size_t found = src.find(MOD_DIR);
+	if (found != std::string::npos) {
+		// Check the path whether be faked.
+		if (src.find(BIN_DIR) == found) {
+			mod_name = src.substr(found+MOD_DIR.length());
+			found = mod_name.find(DIR_DELIM_CHAR);
+			if (found != std::string::npos) mod_name = mod_name.substr(0, found);
+		} else {
+			warningstream << "Mod security:: fake "<< MOD_DIR << " folder found in mod " << src.substr(found+MOD_DIR.length()) << std::endl;
+		}
+	}
+	return mod_name;
+}
+
+std::vector<std::string> inline _get_caller_mod_names(lua_State *L) {
+  lua_Debug info;
+  std::vector<std::string> result;
+  int level = 1;
+	std::string mod_name;
+  while (lua_getstack(L, level, &info)) {
+    FATAL_ERROR_IF(!lua_getinfo(L, "S", &info), "lua_getinfo() failed");
+    mod_name = _get_mod_name_from_debug(&info);
```

Comment:
however there is no need to extract the mod name from the path (??)
the engine knows exactly which mods are loaded and where they reside, you could check against that.

Reply:
I've only found this:

```cpp
        // Only get the mod_name correctly on the top level of `init.lua`
        // Can not be used in functions.
	lua_rawgeti(L, LUA_REGISTRYINDEX, CUSTOM_RIDX_CURRENT_MOD_NAME);
	if (!lua_isstring(L, -1))
		return false;
	std::string mod_name = readParam<std::string>(L, -1);

```